bakery/baking.py

```diff
@@ -231,11 +237,14 @@ async def unbake(
         if not self.is_baked:
             return
 
-        if self.cake_baking_method == BakingMethod.BAKE_FROM_CM:
-            recipe.__exit__(exc_type, exc_value, traceback)
+        # All cakes and piece_of_cakes should already be unbaked
+        # at this moment
+        if not is_cake_or_piece(recipe):
+            if self.cake_baking_method == BakingMethod.BAKE_FROM_CM:
+                recipe.__exit__(exc_type, exc_value, traceback)
 
-        elif self.cake_baking_method == BakingMethod.BAKE_FROM_ACM:
-            await recipe.__aexit__(exc_type, exc_value, traceback)
+            elif self.cake_baking_method == BakingMethod.BAKE_FROM_ACM:
+                await recipe.__aexit__(exc_type, exc_value, traceback)
 
         logger.debug(f"{self} is unbaked")
```

Comment:
We assume that the user will use loguru or logging, he will have to install these dependencies, and otherwise he will fail with an error

Reply:
no dependencies need for bakery. By default `bakery.stuff.DefaultLogger` will be used